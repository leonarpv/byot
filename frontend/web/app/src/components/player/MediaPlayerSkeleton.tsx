import * as React from 'react';
import {Grid, makeStyles, Theme, ThemeProvider} from '@material-ui/core';
import {CSSProperties} from '@material-ui/styles';
import {WithStyles} from '@byot-frontend/web-common/src/types/WithStyles';
import {Patch} from '@byot-frontend/web-common/src/components/elementary/patch/Patch';
import {PatchSkeleton} from '@byot-frontend/web-common/src/components/elementary/patch/PatchSkeleton';
import {theme} from '../../setup';

interface Props extends WithStyles<typeof styles> {}

const styles = (theme: Theme) => ({
  root: {
    width: '100%',
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'flex-end',
    minHeight: 240,
  } as CSSProperties,
  skeleton: {
    height: theme.spacing(2),
  },
});
const useStyles = makeStyles(styles);

export const MediaPlayerSkeleton: React.FC<Props> = (props: Props) => {
  const styles = useStyles(props);
  return (
    <ThemeProvider theme={theme(true)}>
      <Patch classes={{root: styles.root}}>
        <Grid container spacing={1}>
          <Grid item xs={1}>
            <PatchSkeleton variant="rect" />
          </Grid>
          <Grid item xs={1}>
            <PatchSkeleton variant="rect" />
          </Grid>
          <Grid item xs>
            <PatchSkeleton variant="rect" />
          </Grid>
          <Grid item xs={1}>
            <PatchSkeleton variant="rect" />
          </Grid>
        </Grid>
      </Patch>
    </ThemeProvider>
  );
};
