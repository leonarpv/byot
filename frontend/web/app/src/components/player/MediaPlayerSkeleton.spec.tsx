import * as React from 'react';
import {render} from '@testing-library/react';
import {MediaPlayerSkeleton} from './MediaPlayerSkeleton';

describe('<MediaPlayerSkeleton/>', () => {
  it('should render', () => {
    const {container} = render(<MediaPlayerSkeleton />);
    expect(container).toMatchSnapshot();
  });
});
