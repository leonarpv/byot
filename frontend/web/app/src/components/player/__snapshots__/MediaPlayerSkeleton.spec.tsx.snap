// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaPlayerSkeleton/> should render 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-4 makeStyles-root-1 makeStyles-root-3 makeStyles-root-1"
    classes="[object Object]"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
      >
        <span
          class="MuiSkeleton-root makeStyles-root-5 MuiSkeleton-rect MuiSkeleton-pulse"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
      >
        <span
          class="MuiSkeleton-root makeStyles-root-5 MuiSkeleton-rect MuiSkeleton-pulse"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
      >
        <span
          class="MuiSkeleton-root makeStyles-root-5 MuiSkeleton-rect MuiSkeleton-pulse"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
      >
        <span
          class="MuiSkeleton-root makeStyles-root-5 MuiSkeleton-rect MuiSkeleton-pulse"
        />
      </div>
    </div>
  </div>
</div>
`;
